app/src/main/java/com/hanseltritama/connectthree/MainActivity.java

```diff
@@ -1,9 +1,123 @@
+        imageButton.setImageDrawable(getResources().getDrawable(R.drawable.red));
+        imageButton.setAlpha(1f);
+    }
+
+    public void putYellowToken(ImageButton imageButton) {
+        imageButton.setImageDrawable(getResources().getDrawable(R.drawable.yellow));
+        imageButton.setAlpha(1f);
+    }
+
+    public int getXCoord(int imgBtnId) {
+        switch (imgBtnId) {
+            case 0:
+            case 1:
+            case 2: return 0;
+            case 3:
+            case 4:
+            case 5: return 1;
+            case 6:
+            case 7:
+            case 8: return 2;
+            default: return 0;
+        }
+    }
+
+    public int getYCoord(int imgBtnId) {
+        switch (imgBtnId) {
+            case 0:
+            case 3:
+            case 6: return 0;
+            case 1:
+            case 4:
+            case 7: return 1;
+            case 2:
+            case 5:
+            case 8: return 2;
+            default: return 0;
+        }
+    }
+
+    public boolean checkCol(int[][] board, int x_coord) {
+        return (board[0][x_coord] == board[1][x_coord]) && (board[1][x_coord] == board[2][x_coord]);
+    }
+
+    public boolean checkRow(int[][] board, int y_coord) {
+        return (board[y_coord][0] == board[y_coord][1]) && (board[y_coord][1] == board[y_coord][2]);
+    }
+
+    public boolean checkDiagonalLeft(int[][] board, int x_coord, int y_coord) {
+        return (x_coord == y_coord) && ((board[0][0] == board[1][1]) && (board[1][1] == board[2][2]));
+    }
+
+    public boolean checkDiagonalRight(int[][] board, int x_coord, int y_coord) {
+        return (y_coord == 2 - x_coord) && ((board[0][2] == board[1][1]) && (board[1][1] == board[2][0]));
+    }
+
+    public boolean checkWin(int[][] board, int x_coord, int y_coord) {
+        return (checkCol(board, x_coord) || checkRow(board, y_coord) || checkDiagonalLeft(board, x_coord, y_coord) || checkDiagonalRight(board, x_coord, y_coord));
+    }
+
+    public void showAlertDialog(int player, int turn) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this, R.style.Theme_AppCompat_Dialog_Alert);
+        builder.setTitle(turn < 9 ? "Congratulations!" : "Uh-Oh!")
+               .setMessage(turn < 9 ? "Player " + String.valueOf(player) + " Wins!" : "It's a tie game!")
+               .setPositiveButton(R.string.try_again, new DialogInterface.OnClickListener() {
+                   @Override
+                   public void onClick(DialogInterface dialogInterface, int i) {
+                        recreate();
+                   }
+               })
+               .show();
+    }
+
+    public void onBoardClick(View view) {
+        int imgBtnId;
+        int x_coord;
+        int y_coord;
+        ImageButton imageButton = (ImageButton) view;
+        String resEntryName = imageButton.getResources().getResourceEntryName(imageButton.getId());
+
+        imgBtnId = Integer.parseInt(resEntryName.substring(resEntryName.length() - 1));
```

Comment:
button id is just `imageButton.getId()`

Reply:
it returned me some hash numbers with `getId()`

Reply:
my imageButton id is `imageButton0`,...,`imageButton9`